crates/sui-core/src/execution_engine.rs

```diff
@@ -428,6 +428,17 @@ fn pay<S>(
                     Owner::AddressOwner(*recipient),
                     tx_ctx.digest(),
                 );
+                temporary_store.log_event(Event::TransferObject {
+                    package_id: ObjectID::from(SUI_FRAMEWORK_ADDRESS),
+                    transaction_module: Identifier::from(ident_str!("native")),
+                    sender: tx_ctx.sender(),
+                    recipient: Owner::AddressOwner(*recipient),
+                    object_id: new_coin.id(),
+                    version: new_coin.version(),
+                    type_: TransferType::Coin, // Should this be a separate type, like SuiCoin?
+                    amount: Some(amount),
+                });
+
```

Comment:
on L464, this txn also deletes objects, do we want to log events there as well?
on L478, a coin is mutated, another event?

Reply:
yeah we should log DeleteObjects

Reply:
We don't have a `MutatedObject` event today. but i think that will be captured by pat's `CoinBalanceChange`?

Reply:
if we do both txn and object level of logging, then instead of logging objects creation & deletion, we should do that in temp_store and only do txn logging here in exec engine?

Reply:
ya we should do that in temp_store..... I will include the event refactoring in this PR

Reply:
The reason why I thought we should not have MutatedObject as that seems to be too many events, but we could try to justify it I guess.

Reply:
@gegaowp re temp_store.write(), I wonder if that is too low level to put it there .... meaning, the event must have some context around the transaction and its type, and I'm not sure if at that level it knows the context....

Reply:
@velvia the main reason was that, logging object level events in txns would bring some duplicate codes, and it seems pretty standard for deletion & creation.

if the concern is about the events to be emitted, I think either we do not do object level logging, but having creation & deletion,  but not mutation just b/c the extra events it brings, it would be very confusing from pov of the events sub side.

Reply:
One legit use case of MutatedObject is a user/dapp may want to see how an object involves over time. Imagine a weapon in a game, or a very valuable NFT to monitor etc